web/styles/search.css

```diff
@@ -2,150 +2,232 @@
     width: 100%;
     height: var(--header-height);
 
-    .navbar-search:not(.expanded) {
-        display: none;
+    .navbar-search {
+        margin: 5.5px 0;
+        border-radius: 5px;
+        position: absolute;
+        color: hsl(0deg 0% 0% / 50%);
+        background: var(--color-background-search);
     }
 
-    .navbar-search.expanded {
-        float: left;
-        overflow: hidden;
-        margin-top: 0;
-        margin-bottom: 0;
-        width: calc(100% - 2px);
-        position: absolute;
+    .search_icon {
+        text-decoration: none;
+        padding: 0 7px;
+        font-size: 17px;
+        border: none;
+        background-color: transparent;
+        color: hsl(0deg 0% 0%);
+        opacity: 0.5;
+        cursor: default;
 
-        @media (width < $md_min) {
-            width: calc(100% - 42px);
+        &:active,
+        &:focus {
+            outline: none;
         }
 
-        .search_close_button {
-            display: inline;
-            margin-right: 5px;
+        &:disabled {
+            visibility: hidden;
         }
+    }
+
+    .search_close_button {
+        width: 28px;
+        height: 100%;
+        background: none;
+        border: none;
+        opacity: 0.5;
+        line-height: 0;
 
-        .zulip-icon-close {
-            font-size: 15px;
+        &:not(:focus-visible) {
+            outline: none;
+        }
+
+        &:hover {
+            opacity: 1;
+        }
+
+        &:disabled {
+            visibility: hidden;
         }
     }
 
-    .typeahead.dropdown-menu {
-        /* Match the typeahead's width to its parent container. */
+    .zulip-icon-close {
+        font-size: 13px;
+    }
+
+    .navbar-search:not(.expanded) {
         right: 0;
 
+        .search_close_button {
+            display: none;
+        }
+
+        #searchbox-input-container {
+            width: 150px;
+        }
+
+        .search-input {
+            border-radius: 5px;
+            color: var(--color-text-search);
+
+            box-shadow: none;
+            cursor: pointer;
+        }
+
+        &:hover {
+            cursor: pointer;
+
+            @media (width >= $md_min) {
+                box-shadow:
+                    0 4px 20px var(--color-search-box-hover-shadow),
+                    0 1px 5px var(--color-search-box-hover-shadow);
+            }
+
+            .search_icon {
+                cursor: pointer;
+            }
+        }
+
         @media (width < $md_min) {
-            margin-left: 40px;
+            #searchbox-input-container {
+                width: 40px;
```

Comment:
It might be good to put the source of dimensions like this in a comment.

Reply:
I got the dimensions from the Figma file, would you recommend adding that in a comment? That's where I get most of my dimensions.

Reply:
Added a comment here!